crates/cli/src/subcommands/generate/mod.rs

```diff
@@ -120,7 +128,47 @@ Failed to compile module {:?}. See cargo errors above for more details.",
         fs::write(path, code)?;
     }
 
-    format_files(paths, lang)?;
+    format_files(paths.clone(), lang)?;
+
+    if delete_files {
+        let mut files_to_delete = vec![];
+        for entry in fs::read_dir(out_dir)? {
+            let entry = entry?;
+            let path = entry.path();
+            if path.is_file() {
+                if let Ok(contents) = fs::read_to_string(&path) {
+                    if !contents.starts_with("// THIS FILE IS AUTOMATICALLY GENERATED BY SPACETIMEDB.") {
+                        continue;
+                    }
+                }
+
+                if paths
+                    .iter()
+                    .any(|x| x.file_name().unwrap() == path.file_name().unwrap())
+                {
+                    continue;
+                }
+                files_to_delete.push(path);
+            }
+        }
+        if !files_to_delete.is_empty() {
+            println!("The following files were not generated by this command and will be deleted:");
+            for path in &files_to_delete {
+                println!("  {}", path.to_str().unwrap());
+            }
+            println!("Are you sure you want to delete these files? [y/N]");
+            let mut input = String::new();
+            std::io::stdin().read_line(&mut input)?;
+            if input.trim().to_lowercase() == "y" || input.trim().to_lowercase() == "yes" {
```

Comment:
I'd like to see a `--force` flag or similar which skips this check, so that `generate --delete-files --force` can be called in scripts.

Reply:
> I'd like to see a `--force` flag or similar which skips this check, so that `generate --delete-files --force` can be called in scripts.

Added 👍